src/platform/packages/shared/kbn-esql-utils/src/utils/cascaded_documents_helpers.ts

```diff
@@ -497,3 +523,153 @@ export function mutateQueryStatsGrouping(query: AggregateQuery, pick: string[]):
+  // we simply select the commands that exist from start of the query including the insertion anchor command
+  const commandsBeforeInsertionAnchor = ast.commands.slice(0, filterInsertionIndex + 1);
+
+  const filteringWhereCommandIndex = commandsBeforeInsertionAnchor.findLastIndex(
+    (cmd) => cmd.name === 'where'
+  );
+
+  const { operator, expressionType } = getOperator(operation);
+
+  // This is the computed expression for the filter operation that has been requested by the user
+  const computedFilteringExpression =
+    expressionType === 'postfix-unary'
+      ? Builder.expression.func.postfix(operator, [Builder.identifier({ name: fieldName! })])
+      : Builder.expression.func.binary(operator as BinaryExpressionComparisonOperator, [
+          Builder.identifier({ name: fieldName! }),
+          fieldType && isSupportedFieldType(fieldType)
+            ? // @ts-expect-error - this works but needs more type improvements
+              Builder.expression.literal[fieldType].call(
+                null,
+                value as FieldValue<typeof fieldType>
+              )
+            : // when fieldType is not provided or supported, we default to string
+              Builder.expression.literal.string(value as string),
+        ]);
+
+  if (filteringWhereCommandIndex < 0) {
+    const filteringWhereCommand = Builder.command({
+      name: 'where',
+      args: [computedFilteringExpression],
+    });
+
+    mutate.generic.commands.insert(ast, filteringWhereCommand, filterInsertionIndex);
+
+    return BasicPrettyPrinter.print(ast);
+  }
+
+  const filteringWhereCommand = commandsBeforeInsertionAnchor[filteringWhereCommandIndex];
+
+  let modifiedFilteringWhereCommand: ESQLCommand<'where'> | null = null;
+
+  // the where command itself typically only accepts a single expression as its argument
+  // if it's not a named "and" binary expression, we'll treat it as a command that has only one expression,
+  // hence we only need to either replace it with a new one or append a new one
+  if (
+    isBinaryExpression(filteringWhereCommand.args[0]) &&
+    filteringWhereCommand.args[0].name !== 'and'
+  ) {
+    const binaryExpression = filteringWhereCommand
+      .args[0] as ESQLBinaryExpression<BinaryExpressionWhereOperator>;
+    const [left] = binaryExpression.args;
+
+    modifiedFilteringWhereCommand = Builder.command({
+      name: 'where',
+      args:
+        (left as ESQLColumn).name !== fieldName
+          ? [
+              Builder.expression.func.binary('and', [
+                computedFilteringExpression,
+                synth.exp(BasicPrettyPrinter.print(binaryExpression), {
+                  withFormatting: false,
+                }),
+              ]),
+            ]
+          : [
+              // when the expression's left hand's value matches the target field we're trying to filter on, we simply replace it with the new expression
+              computedFilteringExpression,
+            ],
+    });
```

Comment:
Maybe `synth` API can be used here:

```suggestion
    modifiedFilteringWhereCommand = (left as ESQLColumn).name !== fieldName
      ? synth.cmd `WHERE ${computedFilteringExpression} AND ${binaryExpression}`
      : synth.cmd `WHERE ${computedFilteringExpression}`;
```